eras/shelley/impl/testlib/Test/Cardano/Ledger/Shelley/ImpTest.hs

```diff
@@ -232,6 +236,22 @@ import UnliftIO.Exception (
   throwIO,
  )
 
+data SomeSTSEvent era
```

Comment:
The name makes it sound like it's just use for a test, but it's actually something that is part of the testing framework?  I can see it's emitted when submitting an event. Could we find a better name?  What's the purpose of this event?

Reply:
I took inspiration from [`SomeException`](https://hackage.haskell.org/package/base-4.19.1.0/docs/Control-Exception.html#t:SomeException), which comes from the `base` package. The idea here is that we hide the actual type inside the existential so it's "some" STS event which has all the constraints that we expect, but we're not able to find out its actual type.

Reply:
Actually this approach has some problems right now, because you have to be careful when you make an assertion about events, because the actual event might be injected into some other event. We talked with @lehins about making each event a separate datatype so we can get rid of injections, but we'll do that at some other time.